Add Fedora support

jsf9k opened the pull request:
Also note that the scan script has been modified to use `clamscan` instead of `clamdscan`.

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Add support for Fedora

jsf9k committed:
Modify tasks to support Fedora

jsf9k committed:
Modify script to use clamscan instead of clamdscan

jsf9k committed:
Remove clamd service configuration

There is no longer any need to configure, enable, or start clamd since
we are now using clamscan in the cron script.

jsf9k committed:
Add a test to check that the appropriate packages were installed

jsf9k committed:
Blacken test_default.py

jsf9k committed:
Use non-systemd Fedora docker images

This role isn't using systemd when configuring Fedora systems.

jsf9k committed:
Only run systemd tasks on Debian

It isn't necessary to run them on Fedora

Also rely on the more specific "distribution" variable instead of
"os_family".

jsf9k committed:
Tidy up virus_scan.sh

jsf9k committed:
Install clamav instead of clamd for Fedora

clamav provides clamscan, and we aren't using clamd anyway.

jsf9k committed:
Make a cosmetic change to a task name

jsf9k committed:
Add a task to create the /var/log/clamav directory for Fedora

jsf9k committed:
Tidy up README

jsf9k committed:
Remove unused variable

dav3r commented on files/virus_scan.sh:
```diff
@@ -7,7 +7,12 @@ LAST_SCAN_LOG_FILENAME='/var/log/clamav/lastscan.log'
+         --exclude-dir=/dev \
+         --exclude-dir=/sys \
+         --exclude-dir=/proc \
+         /
```
Why are we now excluding these directories from the scanning?  The comment says "scan the entire system" and this is no longer doing that.

jsf9k replied:
Those directories were being excluded via the ClamAV daemon config file before I made any changes.  I had to add those lines to the script to exclude them now that we are using `clamscan` instead of `clamdscan`.

I agree with the code then and now; I don't think it makes sense to scan these particular parts of the root filesystem.  I am happy to change the comment, or remove the exclusions if the group really wants that.

What say you, @felddy and @mcdonnnj?

jsf9k replied:
@dav3r, see 0c49430 for a more accurate comment.

dav3r replied:
I'm good if that was the original behavior, but let's hear from @felddy and @mcdonnnj too.  The updated comment is helpful.

felddy replied:
That's the previous behavior.  See the ansible configuration block that was removed below. 
This is good.

jsf9k committed:
Add more tests

jsf9k committed:
Improve comment

jsf9k commented on tasks/main.yml:
```diff
@@ -1,43 +1,65 @@
-    state: started
+  copy:
+    src: virus_scan.sh
+    dest: /etc/cron.weekly/virus_scan
```
@felddy, is there a reason why we change the script's name when copying it over?  If not then I'd prefer to keep it the same.

felddy replied:
I agree.  Keep it the same.

felddy reviewed (approved):
Looks good.  Thanks for cleaning this up and making it more robust.

dav3r reviewed (approved):
Works for me! Nice work. 🏎

jsf9k committed:
Keep the name when copying the virus scanning script

Modify the tests accordingly.

jsf9k committed:
Correct task name

jsf9k committed:
Update role metadata

mcdonnnj reviewed (approved):
This is a mighty fine looking improvement 👍

jsf9k committed:
Change the metadata "company" field to be CISA Cyber Assessments

This is in accordance with the naming direction handed down from Bobby
Thompson.

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into improvement/add_fedora_support
